pkg/operator/ceph/cluster/controller.go

```diff
@@ -259,17 +264,68 @@ func (r *ReconcileCephCluster) reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, cephCluster, errors.Wrap(err, "failed to add finalizer")
 	}
 
+	// After updating the status of the CephCluster, we need to make sure we have the latest
+	refreshCephCluster := func() error {
+		cephCluster, err = r.context.RookClientset.CephV1().CephClusters(cephCluster.Namespace).Get(context.TODO(), cephCluster.Name, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to get latest CephCluster %q changes", request.NamespacedName.String())
+		}
+		return nil
+	}
+
 	// DELETE: the CR was deleted
 	if !cephCluster.GetDeletionTimestamp().IsZero() {
+		// Set the deleting status
+		opcontroller.UpdateClusterCondition(r.context, cephCluster, request.NamespacedName,
+			cephv1.ConditionDeleting, corev1.ConditionTrue, cephv1.ClusterDeletingReason, "Deleting the CephCluster",
+			true /* keep all other conditions to be safe */)
+		if err := refreshCephCluster(); err != nil {
+			return reconcile.Result{}, cephCluster, err
+		}
+
+		dependents, err := CephClusterDependents(r.context, cephCluster.Namespace)
+		if err != nil {
+			return reconcile.Result{}, cephCluster, err
+		}
+		if !dependents.Empty() {
+			// if there are dependents, we should NOT delete the resource, but we want to keep
+			// reconciling it until it is safe to delete
+			dependentMsg := dependents.StringWithHeader("CephCluster %q will not be deleted until all dependents are removed", r.clusterController.namespacedName.String())
+			logger.Warningf(dependentMsg)
+			r.clusterController.recorder.ReportIfNotPresent(cephCluster, corev1.EventTypeWarning, string(cephv1.ObjectHasDependentsReason), dependentMsg)
+			// Adding the DeletionIsBlocked condition doesn't translate directly to a status/phase,
+			// so do not use the "UpdateClusterCondition" function which will modify that.
+			blockedCondition := buildDeletionBlockedDueToDependentsCondition(dependentMsg)
+			cephv1.SetStatusCondition(&cephCluster.Status.Conditions, blockedCondition)
+			if err := opcontroller.UpdateStatus(r.client, cephCluster); err != nil {
+				return reconcile.Result{}, cephCluster, errors.Wrapf(err, "failed to update CephCluster %q status with %q condition", r.clusterController.namespacedName.String(), blockedCondition.Type)
+			}
+			if err := refreshCephCluster(); err != nil {
+				return reconcile.Result{}, cephCluster, err
+			}
+			return opcontroller.WaitForRequeueIfFinalizerBlocked, cephCluster, fmt.Errorf("waiting to delete")
+		}
+
+		deletingMsg := fmt.Sprintf("deleting CephCluster %q", cephCluster.Name)
+		// Report an event that the cluster is deleting
+		r.clusterController.recorder.ReportIfNotPresent(cephCluster, corev1.EventTypeNormal, string(cephv1.ClusterDeletingReason), deletingMsg)
+		// Remove the DeletionIsBlocked status condition if it exists
+		blockedConditionType := buildDeletionBlockedDueToDependentsCondition("").Type
+		cephv1.RemoveStatusCondition(&cephCluster.Status.Conditions, blockedConditionType)
```

Comment:
Why remove the conditions? The CR will immediately be deleted anyway after the finalizer is removed.

Reply:
Because it is no longer a valid condition. I think it is a good practice, even if it is short-lived. Alternately, it may be good to mark the condition as "False" which I think might be better...